### PR TITLE
feat(elastic): adds stable Elasticsearch Curator dependency

### LINKED ISSUE #11: Add curator as optional deployment
The helm chart should be able to also deploy an curator job like:

**CFG Map**
```
[zell zeebe-benchmark/ ns:zell-helm]$ cat setup/curator-configmap.yaml 
apiVersion: v1
kind: ConfigMap
metadata:
  name: curator-config
  labels:
    app: curator
data:
  action_file.yml: |-
    ---
    # Remember, leave a key empty if there is no value.  None will be a string,
    # not a Python "NoneType"
    #
    # Also remember that all examples have 'disable_action' set to True.  If you
    # want to use this action as a template, be sure to set this to False after
    # copying it.
    actions:
      1:
        action: delete_indices
        description: "Clean up ES by deleting old indices"
        options:
          timeout_override:
          continue_if_exception: False
          disable_action: False
          ignore_empty_list: True
        filters:
        - filtertype: age
          source: name
          direction: older
          timestring: '%Y-%m-%d'
          unit: days
          unit_count: 1
          field:
          stats_result:
          epoch:
          exclude: False
  config.yml: |-
    ---
    # Remember, leave a key empty if there is no value.  None will be a string,
    # not a Python "NoneType"
    client:
      hosts:
        - elasticsearch-master-headless
      port: 9200
      url_prefix:
      use_ssl: False
      certificate:
      client_cert:
      client_key:
      ssl_no_validate: False
      http_auth:
      timeout: 30
      master_only: False
    logging:
      loglevel: INFO
      logfile:
      logformat: default
      blacklist: ['elasticsearch', 'urllib3']
```

**Cronjob**
```
[zell zeebe-benchmark/ ns:zell-helm]$ cat setup/curator-cronjob.yaml 
# https://medium.com/@hagaibarel/running-curator-as-a-kubernetes-cronjob-19eaab9afd3b
apiVersion: batch/v1beta1
kind: CronJob
metadata:
  name: curator
  labels:
    app: curator
spec:
  schedule: "0 3 * * *" # run every 03:00 AM daily
  successfulJobsHistoryLimit: 1
  failedJobsHistoryLimit: 3
  concurrencyPolicy: Forbid
  startingDeadlineSeconds: 120
  jobTemplate:
    spec:
      template:
        spec:
          containers:
          - image: bobrik/curator:5.7.6
            name: curator
            args: ["--config", "/etc/config/config.yml", "/etc/config/action_file.yml"]
            volumeMounts:
            - name: config
              mountPath: /etc/config
          volumes:
          - name: config
            configMap:
              name: curator-config
          restartPolicy: OnFailure
```

### PR DESCRIPTION
**Pull request by npepinpe**
**Description**

Adds curator dependency from the stable Helm charts repository, enabled by default, with a default schedule of every day at 3 am. Configured via `Values.global.elasticsearch-curator`, and can be disabled via `elasticsearch-curator.enabled`.

Closes #11

### CODE REVIEW

**Comment by salaboy** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **npepinpe**

If they are not already assigned, you can assign the PR to them by writing `/assign @npepinpe` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=zeebe-io%2Fzeebe-cluster-helm).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/zeebe-io/zeebe-cluster-helm/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by npepinpe** [comment]
I thought about using the same condition for Elastic and curator, as you probably want both - but it looks like later versions of Elastic solved the issue and do not need curator anymore I think with the new snapshot support, so maybe it's better to leave them separate for now.

/ok-to-test

File: README.md [hunk 1]
```diff
@@ -31,6 +31,8 @@ This functionality is in beta and is subject to change. The design and code is l
 | `annotations`                 | annotations to be applied to the StatefulSet and Service                                                                                                                                | ``                                                                                                           |
 | `podAnnotations`                 | annotations to be applied to the StatefulSet pod Template                                                                                                                                | ``                                                                                                           |
 | `elasticsearch.enabled`                 | Enable ElasticSearch deployment as part of the Zeebe Cluster                                                                                                                                | `true`                                                                                                           |
+| `elasticsearch-curator.enabled`         | Enables the Elasticsearch curator to regularly clean up your Elastic indexes
+| `true`
```
Comment (Zelldon) [inline README.md]: Lets not enabled it by default, because this will delete users data which they might not expect if they just update there helm charts

**Comment by npepinpe** [comment]
I minimally tested it, as I expect curator to just work. So I installed the chart and could see curator created with the right configuration, but I didn't test that its actual functionality. I'll do that with a small schedule :+1:

**Comment by npepinpe** [comment]
Doesn't quite work - it doesn't complain oddly enough, but yeah you were correct, it doesn't pick up the right host. This chart, although official, is not easily customizable - maybe we do end up setting up curator manually =/

**Comment by salaboy** [comment]
@npepinpe this is one of the cases where we can contribute back to the chart if it is broken in some way.. usually, chart maintainers don't expose all the things that they can, and just wait for community members to provide PRs if further parameterization is needed. 
Any ideas of what is missing there?

**Comment by npepinpe** [comment]
It's not missing so much as I don't know how to reuse our `elasticsearch.host` and `elasticsearch.port` values to pass them on to the other chart. Any ideas?

The curator chart expects us to generate a `config_yaml` value which is just, well, raw YAML to use for their config file.

**Comment by npepinpe** [comment]
Closing as I really don't remember this, and it wasn't working so we better look into this another time :+1:

**Pull request closed without merging**